b1ddi/resource_ipam_address.go

```diff
@@ -180,57 +184,31 @@ func resourceIpamsvcAddressCreate(ctx context.Context, d *schema.ResourceData, m
 		}
 	}
 
-	if d.Get("address").(string) != "" {
-
-		a := &models.IpamsvcAddress{
-			Address:   swag.String(d.Get("address").(string)),
-			Comment:   d.Get("comment").(string),
-			Host:      d.Get("host").(string),
-			Hwaddr:    d.Get("hwaddr").(string),
-			Interface: d.Get("interface").(string),
-			Names:     names,
-			Parent:    d.Get("parent").(string),
-			Range:     d.Get("range").(string),
-			Space:     swag.String(d.Get("space").(string)),
-			Tags:      d.Get("tags"),
-		}
-
-		resp, err := c.IPAddressManagementAPI.Address.AddressCreate(&address.AddressCreateParams{Body: a, Context: ctx}, nil)
-		if err != nil {
-			return diag.FromErr(err)
-		}
-
-		d.SetId(resp.Payload.Result.ID)
+	addressStr := d.Get("address").(string)
+	if !strings.HasPrefix(addressStr, "ipam") {
+		addressPath = addressStr
 	} else {
-		resp, err := c.IPAddressManagementAPI.Subnet.SubnetCreateNextAvailableIP(&subnet.SubnetCreateNextAvailableIPParams{
-			ID:      d.Get("parent").(string),
-			Context: ctx,
-		}, nil)
-		if err != nil {
-			return diag.FromErr(err)
-		}
-
-		body := &models.IpamsvcAddress{
-			Address:   resp.Payload.Results[0].Address,
-			Comment:   d.Get("comment").(string),
-			Host:      d.Get("host").(string),
-			Hwaddr:    d.Get("hwaddr").(string),
-			Interface: d.Get("interface").(string),
-			Names:     names,
-			Range:     d.Get("range").(string),
-			Tags:      d.Get("tags"),
-		}
+		addressPath = fmt.Sprintf("%s/%s", addressStr, NAIP_PATH)
+	}
```

Comment:
Same comment as in the subnet PR. We shouldn't be recycling the "address" field for this. Ideally the "parent" field would have been a good fit. But it seems that for next available IP, if we provide an address block ID, the parent can be any subnet in the address block as well. And this would cause the same problem with having a non empty plan. 

Maybe the ideal thing to do here, (and in subnet) is to have a separate field in terraform `next_avaiable
_in` (or any other name), and use that to pass the container ID for next available generation.

Reply:
@akleinib @mathewab and I had a discussion offline and have determined using a `next_available_id` to be defined for both Next available Subnet and IP. If this field is defined, it is a next available resource call.

Also, in the Next Available Sunet resource implmentation, we can call the endpoint `/ipam/subnet/<id>/nextavailablesubnet` to create the subnet using the SubnetCreate API. I will be updating this in next updates to this PR.